R/adjective_animal.R

```diff
@@ -64,5 +73,50 @@ adjective_animal <- function(n = 1, n_adjectives = 1, style = "snake",
     gfycat_animals <- gfycat_animals[nchar(gfycat_animals) <= max_len[2]]
   }
   vals <- c(rep(list(gfycat_adjectives), n_adjectives), list(gfycat_animals))
-  ids(n, vals = vals, style = style)
+  if (alliterate) {
+    aa_alliterate(n, vals, style)
+  } else {
+    vals <- c(rep(list(gfycat_adjectives), n_adjectives), list(gfycat_animals))
+    ids(n, vals = vals, style = style)
+  }
+}
+
+
+## We can generate alliterative ids by either rejection sampling
+## (which will be hard with multiple adjectives) or by doing a
+## weighted sample of letters and then working with each letter
+## separately.  Do do this properly we should compute the number of
+## distinct combinations and avoid duplications but that seems
+## excessive for this and is only an issue if the number of adjectives
+## is greater than one.
```

Comment:
Discussion point: when I tested this:
```
> ids::adjective_animal(alliterate = TRUE, max_len=3, n_adjectives=3)
[1] "fat_fat_fit_fly"
```
I almost always got adjectives beginning with F - I guess because there are more 3-lettered, um, F-words, than there are others. (a=2, b=2, c=3, d=2, e=1, f=7, g=1, h=2, i=3, j=1, k=1, l=2, m=2, n=1, o=3, p=0, r=2, s=4, t=3, u=0, v=0, w=3, x=0, y=0, z=0). 

This is a weaknes of the adjective set, and indeed, the English language, but not of the algorithm I think.

Reply:
Maybe a future extension could include a parameter for `forbid_duplicated_adjectives = TRUE`